contents/security/crossSiteScripting/crossSiteScripting.md

```diff
@@ -9,64 +9,87 @@
 
-## Overview
+## What is XSS?
+Cross Site Scripting (XSS) is the most exploited web application vulnerability in 2017 (based on a <a href="https://www.ptsecurity.com/upload/corporate/ww-en/analytics/Web-application-attacks-2018-eng.pdf" target="_blank">report</a> in 2018). XSS vulnerabilities have been reported and exploited since the 1990s. Prominent sites affected in the past include social-networking sites <a href="https://www.symantec.com/connect/blogs/persistent-xss-vulnerability-facebook" target="_blank">Facebook</a> and <a href="https://www.acunetix.com/blog/articles/dangerous-xss-vulnerability-found-on-youtube-the-vulnerability-explained/" target="_blank">Youtube</a>.
 
-Cross site scripting (XSS) is a vulnerability found in websites.
+XSS is a type of injection, in which malicious code are injected into trusted websites. XSS flaws that allow these attacks to happen are widespread and can occur anywhere a web application uses input from a user to generate an output on the website without validating it or encoding it. 
 
-Text boxes are commonly used in forms in websites to ask the user for input. For
-example, blog websites may provide a text box for visitors of the blog to enter comments.
+## Why do you need to know about XSS?
+As software engineers, or aspiring software engineers, it is highly likely that we will build web applications. It is important to know how we can protect our web applications against XSS, so that it will not become a tool for attackers.
 
-However, are commenters limited to just typing in plain text? Some websites do not
-check the input, thus commenters are able to use formatting HTML tags, such
+## How does XSS work?
+One of the most common ways to accept inputs from users are text boxes. In this example, we will look at a website that allows users to enter their comments to blog posts. 
+
+Here is the comments section of the website: 
+
+> **Comments Section** <br>
+That's a very nice picture! <br>
+Good photograph!
+
+However, are users limited to just typing in text? This website does not check the input, thus users are able to use formatting HTML tags, such
 as `<b>` for bolding and `<i>` for italicizing.
 
+What happens when a user submits the following as a comment?
 ```html
-I <b>like</b> your blog post!
+I <b>like</b> your photograph!
 ```
 
-becomes: I **like** your blog post!
+It becomes: 
+> **Comments Section** <br>
+That's a very nice picture! <br>
+Good photograph! <br>
+I **like** your photograph!
+
+HTML also supports <trigger for="modal:script">`<script>`</trigger>, for you to write `JavaScript` code on the webpage as well. 
 
-The problem arises when the commenter starts using HTML for malicious purposes. For
-example, HTML also supports `<script>`, for you to write JavaScript code on the
-webpage as well.
+<modal title="**`<script>` and `</script>` in html**" id="modal:script">
+In HTML, anything between the opening and closing script tags will run as <b>JavaScript</b>. 
+</modal>
```

Comment:
A popover or a tooltip is enough here? If it is a modal, make it appear on a click, not hover.

Reply:
Changed all the modals to tooltips